lldb/tools/lldb-dap/src-ts/extension.ts

```diff
@@ -15,13 +13,14 @@ import { DisposableContext } from "./disposable-context";
 export class LLDBDapExtension extends DisposableContext {
   constructor() {
     super();
+    const factory = new LLDBDapDescriptorFactory();
+    this.pushSubscription(factory);
```

Comment:
What does this do?

Reply:
I made the factor itself implement `vscode.Disposable` so if the extension is deactivated it will stop the server, if its running.